### PR TITLE
Ruby - x-ms-parameter-grouping for Ruby & Azure.Ruby Generator

### PR DESCRIPTION
**Pull request by vishrutshah**
Implementing x-ms-paramete-grouping swagger extension for Ruby & Azure.Ruby generator

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: AutoRest/Generators/Ruby/Ruby.Tests/RspecTests/parameter_grouping_spec.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    first_parameter_group.header_one = @header
+
+    second_parameter_group.query_two = 42
+
+    result = @client.parameter_grouping.post_multi_param_groups_async(first_parameter_group, second_parameter_group).value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should allow parameter group objects to be shared between operations' do
```
Comment (veronicagg) [inline AutoRest/Generators/Ruby/Ruby.Tests/RspecTests/parameter_grouping_spec.rb]: in this case, shouldn't we be creating the FirstParameterGroup grouping once and pass it to 2 different operations, so test the "sharing" of the parameter group? If not, it looks like this test is validating the same as some of the above.

File: gulpfile.js [hunk 2]
```diff
@@ -86,7 +86,8 @@ var rubyMappings = {
   'http_infrastructure':['../../../TestServer/swagger/httpInfrastructure.json','HttpInfrastructureModule'],
   'required_optional':['../../../TestServer/swagger/required-optional.json','RequiredOptionalModule'],
   'report':['../../../TestServer/swagger/report.json','ReportModule'],
-  'model_flattening':['../../../TestServer/swagger/model-flattening.json', 'ModelFlatteningModule'],
+  'model_flattening':['../../../TestServer/swagger/model-flattening.json', 'ModelFlatteningModule'],
```
Comment (veronicagg) [inline gulpfile.js]: do we need to add this one in this PR?

Comment (vishrutshah) [inline gulpfile.js]: I see that it was already there but i am not sure why gitdiff shows it as removed and then added back. I guess that new line character was added after line 89 so it shows that as modified.

**Comment by veronicagg** [comment]
@vishrutshah We chatted about the above comments, rest looks good to me. Thanks!

**Comment by vishrutshah** [comment]
Thanks for the review !!

**Pull request merged**